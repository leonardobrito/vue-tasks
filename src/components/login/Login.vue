<template>
  <div>
    <h1>Login</h1>
    <div>
      <label>Username:</label>
      <input type="text" v-model="username" />
    </div>
    <div>
      <label>password:</label>
      <input type="password" autocomplete="new-password" v-model="password" />
    </div>
    <button @click="handleDoLogin(username, password)">Login</button>
  </div>
</template>

<script>
import { doLoginApi } from "../../services/api";
import { mapActions, mapGetters } from "vuex";

export default {
  data() {
    return {
      username: "",
      password: ""
    };
  },
  computed: {
    ...mapGetters('login', ['getToken']),
  },
  watch: {
    getToken(newToken) {
      if(newToken) this.redirectToList()
    }
  },
  methods: {
    ...mapActions('login', ['doLogin']),
    handleDoLogin(username, password) {
      const payload = { username, password };

      this.doLogin(payload);
    },
    redirectToList(){
      this.$router.push({ name: "listTaskgroup" });
    }
  }
};
</script>

<style></style>
