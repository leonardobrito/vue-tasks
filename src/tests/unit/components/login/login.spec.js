import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Login from '@/components/login/Login.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Components > Login', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Login, {
      localVue,
      store: new Vuex.Store({
        modules: {
          'login': {
            namespaced: true,
            getters: {
              getToken: () => null
            }
          }
        }
      })
    })
  });

  it('should matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
