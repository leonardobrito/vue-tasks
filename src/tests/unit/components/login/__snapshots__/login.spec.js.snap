// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Login should matches snapshot 1`] = `
<div>
  <h1>
    Login
  </h1>
   
  <div>
    <label>
      Username:
    </label>
     
    <input
      type="text"
    />
  </div>
   
  <div>
    <label>
      password:
    </label>
     
    <input
      autocomplete="new-password"
      type="password"
    />
  </div>
   
  <button>
    Login
  </button>
</div>
`;
